fe/fe-core/src/main/java/com/starrocks/qe/ShortCircuitHybridExecutor.java

```diff
@@ -193,17 +196,17 @@ private SetMultimap<TNetworkAddress, TabletWithVersion> assignTablet2Backends()
         scanRangeLocations.forEach(range -> {
             ImmutableMap<Long, Backend> idToBackend =
                     GlobalStateMgr.getCurrentState().getNodeMgr().getClusterInfo().getIdToBackend();
+            List<Backend> aliveBackends =
+                    idToBackend.values().stream().filter(be -> isWorkerAvailable(be)).collect(Collectors.toList());
+            Collections.shuffle(aliveBackends);
 
             TInternalScanRange internalScanRange = range.getScan_range().getInternal_scan_range();
             TabletWithVersion tabletWithVersion = new TabletWithVersion(internalScanRange.getTablet_id(),
                     internalScanRange.getVersion());
 
-            TNetworkAddress be = pick(internalScanRange.getHosts());
-            idToBackend.forEach((id, backend) -> {
-                if (backend.getHost().equals(be.getHostname()) && (backend.getBePort() == be.getPort())) {
-                    backend2Tablets.put(new TNetworkAddress(be.getHostname(), backend.getBrpcPort()), tabletWithVersion);
-                }
-            });
+            Optional<TNetworkAddress> be = pick(internalScanRange.getHosts(), aliveBackends);
+            be.ifPresent(backend -> backend2Tablets.put(new TNetworkAddress(backend.getHostname(), backend.getPort()),
```

Comment:
When there isn't available backend, should throw an exception with the error message `Compute node not found. Check if any compute node is down. backend: [...]`. Could see `DefaultWorkerProvider::reportWorkerNotFoundException` for some inspiration.